AppCenter/AppCenter/Internals/Context/UserId/MSUserIdContext.m

```diff
@@ -79,7 +79,7 @@ - (NSString *)userId {
 - (void)setUserId:(nullable NSString *)userId {
   NSArray *synchronizedDelegates;
   @synchronized(self) {
-    BOOL sameUserId = (!userId && !self.currentUserIdInfo.userId) || [self.currentUserIdInfo.userId isEqualToString:(NSString *)userId];
+    BOOL sameUserId = (!userId && !self.currentUserIdInfo.userId) || (userId && [self.currentUserIdInfo.userId isEqualToString:(NSString *)userId]);
```

Comment:
Does this change the logic? Has this part been tested?

Reply:
As I understand, _isEqualToString_ returns _true_ only when both parameters are not _nil_ and equals each other. 
A case when both are _nil_ considered in the first part. The second part returns _true_ for the same cases as it was before (both _non-nil_ and equals each other).